### DISCUSSION TITLE
v5: initial impressions ( :-) 👍  ), and ideas for future versions

### DISCUSSION THREAD

**Discussion by Elementari** [mbnuqw/sidebery#1257] [Ideas]
Hello!!!

Thank you so much for the v5 release - especially the search bar!!!

I spent 2 hours configuring and I'm very happy now - I did some major tweaks: special mention of the wonderful Style Editor!!! Beautiful!!! 

THANK YOU AGAIN!

I wanted to share some small things that I was missing - I'm packing these into this post so as to not flood the forum.

1. Search Bar
1.1 Idea: A button to toggle the search bar would be great, it could be located in the bar above the search bar, where the panel buttons are etc. 
1.2 Idea: When a search is active, it would be great if the search bar were highlighted e.g. in bright red to make obvious to the user that the tabs below are only a selection. (In Directory Opus, I have even configured so that everything (!) is colored, i.e. in the case of sideberry, all tabs (except the active tab) would then change color to indicate they're filtered.

2. Idea: Checkbox mode to select tabs (in order to: move them to new window, unload them, close them, pin them, move them to a new panel, move them to bottom of tabs list, etc.)

3. Bugs/Issues? 
3.1 Bug/Issue? -> Pinned tabs: I don't know if this is a bug, but I remember all my pinned tabs used to load upon FF start, now only three of them load. "limit amount of simul. loading tabs" is set to 10, "prevent unloading of pinned tabs" is ON.
edit: browser.sessionstore.restore_pinned_tabs_on_demand is *false*
3.2 Bug/Issue? -> "limit amount of loading tabs": disabling this setting is not memorized. it's also not quite clear what this would entail.
3.3 Bug/Issue? -> The interface is a mix of German and English -- I would prefer it to be completely in English - a "choose interface language" button would be best I think.
3.4 Bug/Issue: -> Tab count font is not readable when height of row/icon is changed - the tab count font should have an opaque background
3.5 Bug/Issue: -> Loss of ability to handle vast amounts of tabs; either because of amount of loaded tabs (in memory), OR because of very high numbers of unloaded tabs (greyed out): (First of all, a trigger warning for tab phobics). -> I had 2500 (two thousand five hundred) tabs open in FF (most of them unloaded) (MBA M1 16GB, and Win11 VM running...) and had no issues whatsoever with Sideberry 4 (smooth as ever), but Sideberry 5 had major issues: lagging, unreactive, massive rendering delays (i.e., blank areas, including in the main website window of SB settings. (But no tabs were lost.) I also restarted FF, issue persisted.

4. Idea: New tab creation on double-click blank panel bar space: I was thinking perhaps it would be a nice feature that double-clicking on unused space of the panel bar would create a new tab (the bar below the large FF sidebar bar where it says "Sideberry" as a default).

5. Idea: Alternatives for users with not middle mouse button: Many users don't have a middle mouse button -> features such as "unload tab" could be made available by right-clicking favicon of a tab, for example.

**Comment by zelch** [comment]
For 3.5, disable 'Hide native tabs of inactive panels', and possible 'Hide native tabs in folded branch'.

I did some profiling after getting frustrated with the lag issues when switching panels, and it turns out that Firefox has performance issues when being told to hide or unhide a thousand tabs.

For 2, you can select ranges of tabs by holding down shift, or select individual tabs by holding down control.

For 1.1, while not exactly what you're asking for, if you set the search bar to Dynamic, hit the x that appears on the right of the search bar to close it, and then bind something to search, then when you use the binding the search bar appears. (And is focused.)  When you're done searching, the search bar goes away again.

I hope these help. :)

**Comment by Elementari** [comment]
thank you.

3.5: "Hide native tabs of inactive panels" is already disabled. The second one you mention I cannot find, but any and all settings that contain hide/verstecken and "native/nativ" are disabled.

2. yes, thank you. I knew that, I just like checkboxes: In DOpus, one can go through a list of files with the cursors (up down arrow keys) and select them with space (no mouse, no constant pressing of shift). It's just very comfortable (and also not prone to wrong clicks that might cause the selection to be lost). I should have mentioned these advantages. It seems interesting for selecting 10 tabs that are randomly scattered amoung 1000 tabs, and for selecting lots of tabs comfortably. In DOpus, you can also e.g. select 5 items with checkboxes. then mark 10 other files that are all below each other by using shift-arrow, and then check them with one spacebar-bress, in addition to the 5 already checked.

1.1: yes, I saw this, thank you. I'm not a big fan of keyboard shortcuts for extensions, but sure. It's not a big deal - I just thought perhaps a search bar toggle button might not be that difficult to make, and the other buttons also theoretically could be non-existent because one could also bind them all to keys. So it's only logical that search would also get a button as an option. Just a suggestion.

**Comment by zelch** [comment]
On 3.5, I _believe_ that the German translation for the `Hide native tabs in folded branch` is `Verstecke eingeklappte Tabs`.

On 2, yeah, I did some poking at the options available, and I agree that the current set is definitely suboptimal.

I _think_ that we need 3 'selection' state bits for tabs, but please correct me if I'm wrong here, and really, they are 

* Active: The tab that's being displayed right now.
* Selected: The tab is selected.
* Cursor: If you use the 'Select up' and 'Select down' bindings in Sideberry today, that does something remarkably close to this, but since it's doing it with selection, it breaks almost everything when trying to do what I think @Elementari is trying to do.

We would then have bindings to move the cursor, to toggle selection on the tab the cursor is on, and then 'select range up' and 'select range down' would be relative to the cursor position, not the active tab, and importantly, would _not_ reset the selection state of any other tabs. 

Does that sound right?

**Comment by Elementari** [comment]
1.1 does work. one just has to add the search button to the panel. :-)

**Comment by Elementari** [comment]
3.3 is not a huge issue because it's possible to change FF interface to English completely. (unless you don't understand English)

**Reply by zelch** [reply]
For 3.3, If you're interested, you can clone the repository and look at the files in src/_locales/

Any that are missing a de string will show up in English, and if you put in German translations in de, I am quite sure that the translations would be welcomed.

**Comment by Elementari** [comment]
OFF TOPIC >

I would have done this this if ChatGPT wasn't so unreliable.  I gave up after some tries.
I'm a German native speaker and I'm able to do precise translations to German, and in particular I can check if DeepL / ChatGPT translations are correct.
If I were given a list that looks like this, I would check every single translation:
EN text
DE Text (current translation if available, I will fix these too if problematic)
DE ChatGPT suggestion
DE DeepL suggestion (optional)
DE (FINAL) (I will fill in)

That said, my opinion is that the subset of German speakers who 
1) use FF
2) use sideberry
3) spend time configuring it
4) can't read English at least to the degree that only a couple of words would need to be thrown at a translator
is so small it's not really worth it to even keep the German translation. 

In fact the amount of people upset because of the mix of German and English (which makes it difficult to CTRL-F for keywords) might be larger than the subset described above....

< END OFF TOPIC

**Comment by Elementari** [comment]
I'm adding another "issue" that I'm encountering: Once in a while I drag a tab by mistake, causing a new window to be created. -> There seems to be no option to disable this behavior completely, i.e. such that NOTHING happens when a tab is cragged outside of the sidebar.

**Reply by zelch** [reply]
Settings -> Drag and Drop -> Action on drag-and-drop to the outside of sidebar.

Set it to 'Pass dragged data to the target', and you'll stop getting new windows being created unless you really want them.

**Reply by Elementari** [reply]
Hello - No, this does not work and is actually even worse. It will override the currently displayed webpage.
E.g.
Tab 1: amazon.com (active)
Tab 2: microsoft.com

---> Drag microsoft to the right
---> you will have 2x microsoft, and amazon is gone. (poor innocent amazon, evil microsoft)

**Reply by albino1** [reply]
I don't know if this would fix your specific problem, but I use an addon called Disable Tab Detach to stop new windows from being created when I accidentally drag tabs:

https://addons.mozilla.org/en-US/firefox/addon/disable-tab-detach/

**Reply by Elementari** [reply]
First of all, that's really friendly of you all, with all your suggestions and help!
I will consider this solution, albino! -- Nonetheless, I think a "Disabled" option for the drag action would still be helpful for many users, and possiblty not difficult to implement (?).
